lib/launchers/openstack10.rb

```diff
@@ -822,6 +824,34 @@ def get_routers
       return res[:parsed]["routers"]
     end
 
+    def get_instances
+      get_objects("servers")
+    end
+
+    def get_instance_detail(id)
+      res = self.rest_run(self.os_compute_url + "/servers/#{id}", "GET", {}, self.os_token)
+      raise res[:response] unless res[:success]
+      return res[:parsed]
+    end
+
+    def get_running_instances
+      threads = []
+      running_inst = {}
+      instances = get_objects("servers")
+      instances.each do |instance|
+        threads << Thread.new(instance) do | i|
```

Comment:
better throttle this to 15 or so connections

Reply:
use method `array#each_slice`

Reply:
@akostadinov what do you mean by using #each_slice?   We probably don't need to throttle as Jenkins only returns 100 entries max.  Also, if we do want to use threadpool, then that's another gem we need to install

Reply:
100 connections at the same time not very little but sounds not terrible.
wrt a thread pool, we can jst create that number of threads per each slice. Not needed to use a thread pool.